recipes-ni/ni-test-boot-time/files/upload_results.py

```diff
@@ -0,0 +1,137 @@
+    try:
+        process = subprocess.run(['uname', '-r'],
+                                 check=True,
+                                 stdout=subprocess.PIPE)
+        kver_full = process.stdout.strip().decode("utf-8")
+        rgx = re.search(r'([0-9]+\.[0-9]+)\.([0-9]+)', kver_full)
+        kver = rgx.group(1)
+    except (FileNotFoundError, AttributeError):
+        logging.warning("failed to read the current kernel version")
+        raise
+
+    return kver, kver_full
+
+
+def get_os_version():
+    """Extract OS version information from /etc/os-release"""
+    try:
+        with open('/etc/os-release', 'rb', 0) as file, \
+             mmap.mmap(file.fileno(), 0, access=mmap.ACCESS_READ) as mfile:
+            rgx = re.search(br'VERSION_ID=([0-9.]+)', mfile)
+            version = rgx.group(1).decode()
+
+            rgx = re.search(br'BUILD_ID="([^"]+)"', mfile)
+            build = rgx.group(1).decode()
+
+            rgx = re.search(br'VERSION_CODENAME="([^"]+)"', mfile)
+            codename = rgx.group(1).decode()
+    except (ValueError, AttributeError):
+        logging.warning("failed to parse version information from '/etc/os-release'")
+        raise
+
+    return version, build, codename
+
+
+def get_boot_time(log_file):
+    """Extract boot time from log file"""
+    try:
+        with open(log_file, 'rb', 0) as file, \
+             mmap.mmap(file.fileno(), 0, access=mmap.ACCESS_READ) as mfile:
+            rgx = re.search(br'([0-9.]+)', mfile)
+            boot_time = float(rgx.group(1).decode())
+    except (FileNotFoundError, ValueError, AttributeError):
+        logging.warning("failed to parse boot time from '%s'", log_file)
+        raise
+
+    return boot_time
+
+
+def read_data(path):
+    """Compile test data and metadata"""
+    controller = get_device()
+    kernel_version, kernel_full_version = get_kernel_version()
+    distro_version, distro_build, distro_name = get_os_version()
+    boot_time = get_boot_time(path)
+
+    print("boot time:", boot_time, "(s)")
+
+    data = {
+        "measurement": "boot_performance",
+        "tags": {
+            "controller": controller,
+            "kernel_version": kernel_version,
+            "kernel_full_version": kernel_full_version,
+            "distro_version": distro_version,
+            "distro_build": distro_build,
+            "distro_name": distro_name,
+        },
+        "fields": {
+            "boot_time": boot_time,
+        },
+        "time": str(datetime.now())
+    }
+    return data
+
+
+def upload_results(data, server, server_port):
+    """Upload results to influxdb instance"""
+    try:
+        influxdb = InfluxDBClient(host=server, port=server_port)
+        influxdb.switch_database("rtos_boot_performance")
+        if not influxdb.write_points(data):
+            logging.warning("failed to write data points to influxdb")
+        influxdb.close()
+    except (exceptions.InfluxDBClientError,
```

Comment:
I believe an exception here will not trigger the ptest to fail since you are catching and not re-throwing. I don't know if that's what you intend. Since at this point you will have found influxdb info in /home/admin/.influxdb.info, maybe it should cause a failure?